File: BE-Task_1-Php/internal/domain/entities/agent.php
<?php

declare(strict_types=1);

namespace App\Internal\Domain\Entities;

use App\Internal\Domain\ValueObjects\AgentRole;
use App\Internal\Domain\ValueObjects\Region;

final readonly class Agent
{
    public function __construct(
        public string $agentId,
        public AgentRole $role,
        public Region $region,
    ) {
    }

    public function toArray(): array
    {
        return [
            'agent_id' => $this->agentId,
            'role' => $this->role->value,
            'region' => $this->region->value,
        ];
    }
}

File: BE-Task_1-Ruby/internal/domain/entities/agent.rb
module RefundBackend
  module Domain
    module Entities
      class Agent
        attr_reader :agent_id, :region, :role

        def initialize(agent_id:, role:, region:)
          @agent_id = agent_id
          @role = role
          @region = region
          freeze
        end

        def to_h
          {
            agent_id: agent_id,
            role: role,
            region: region
          }
        end
      end
    end
  end
end
